src/diskquota.c

```diff
@@ -163,6 +166,40 @@ diskquota_launcher_shmem_size(void)
 	size = add_size(size, mul_size(MAX_NUM_MONITORED_DB, sizeof(struct DiskquotaDBEntry))); // hidden memory for dbArray
 	return size;
 }
+
+static bool
+is_altering_extension(void)
+{
+	if (ActivePortal == NULL) return false;
+	/* QD: When the sourceTag is T_AlterExtensionStmt, then return true */
+	if (ActivePortal->sourceTag == T_AlterExtensionStmt) return true;
```

Comment:
But we don't return false on QD otherwise. As a result, we anyway validate `sourceText` that may contain any garbage.